ivre/db/mongo.py

```diff
@@ -904,6 +904,28 @@ def searchfile(self, fname):
             output=fname,
         )
 
+    def searchsmbshares(self, access=None):
+        """Filter SMB shares with anonymous `access` (default: either
+        read or write, accepted values 'r', 'w', 'rw')
+
+        """
+        if access is None:
+            access = ''
+        access = access.lower()
+        if 'w' in access:
```

Comment:
This code snippet is quite hard to understand. You could set the default value of `access` to `re.compile('^(READ|WRITE)')` and then apply a simpler test.
Or use a dictionnary with `access` as a key.